src/promptflow-evals/tests/evals/unittests/test_evaluate.py

```diff
@@ -61,3 +95,37 @@ def test_evaluate_missing_required_inputs(self, missing_columns_jsonl_file):
             evaluate(data=missing_columns_jsonl_file, evaluators={"g": F1ScoreEvaluator()})
 
         assert "Missing required inputs for evaluator g : ['ground_truth']." in exc_info.value.args[0]
+
+    def test_evaluate_missing_required_inputs_target(self, questions_wrong_file):
+        with pytest.raises(ValueError) as exc_info:
+            evaluate(data=questions_wrong_file,
+                     evaluators={"g": F1ScoreEvaluator()},
+                     target=_target_fn
+                     )
+        assert "Missing required inputs for target : ['question']." in exc_info.value.args[0]
+
+    def test_wrong_target(self, questions_file):
+        """Test error, when target fuction does not generate required column."""
+        with pytest.raises(ValueError) as exc_info:
+            # target_fn will generate the "answer", but not ground truth.
+            evaluate(data=questions_file, evaluators={"g": F1ScoreEvaluator()}, target=_target_fn)
+
+        assert "Missing required inputs for evaluator g : ['ground_truth']." in exc_info.value.args[0]
+
+    @pytest.mark.parametrize('script_is_file', [True, False])
+    def test_save_fun_as_flow(self, tmpdir, pf_client, script_is_file):
+        """Test saving function as flow."""
+        with patch('promptflow.evals.evaluate._utils.os') as mock_os:
+            mock_os.path.isfile.return_value = script_is_file
+            save_function_as_fow(_target_fn, tmpdir, pf_client)
+        assert os.path.isfile(os.path.join(tmpdir, 'flow.flex.yaml'))
+
+    def test_apply_target_to_data(self, pf_client, questions_file, questions_answers_file):
+        """Test that target was applied correctly."""
+        qa = _apply_target_to_data(_target_fn, questions_file, pf_client)
+        results = pd.read_json(qa, lines=True)
+        ground_truth = pd.read_json(questions_answers_file, lines=True)
+        try:
+            assert_frame_equal(results, ground_truth, check_like=True)
+        finally:
+            os.unlink(qa)
```

Comment:
I think these are more like unit-test instead of e2e tests.

Reply:
Do we keep the unit tests in different file?

Reply:
I think it's fine to leave them here since they are evaluate api related.

Reply:
Moved.